src/display/lv_display.c

```diff
@@ -398,11 +398,29 @@ void lv_display_set_draw_buffers(lv_display_t * disp, lv_draw_buf_t * buf1, lv_d
 void lv_display_set_buffers(lv_display_t * disp, void * buf1, void * buf2, uint32_t buf_size,
                             lv_display_render_mode_t render_mode)
 {
+    LV_ASSERT(buf1 != NULL);
+    if(buf1 == NULL) {
+        LV_LOG_ERROR("buf1 is NULL");
+        return;
+    }
+
+    lv_color_format_t cf = lv_display_get_color_format(disp);
     uint32_t w = lv_display_get_horizontal_resolution(disp);
     uint32_t h = lv_display_get_vertical_resolution(disp);
     LV_ASSERT(w != 0 && h != 0);
 
-    lv_color_format_t cf = lv_display_get_color_format(disp);
+    if(buf1 != lv_draw_buf_align(buf1, cf)) {
+        /* buf1 is not aligned according to LV_DRAW_BUF_ALIGN */
+        LV_LOG_ERROR("buf1 is not aligned");
+        return;
+    }
+
+    if(buf2 && buf2 != lv_draw_buf_align(buf2, cf)) {
+        /* buf2 is not aligned according to LV_DRAW_BUF_ALIGN */
+        LV_LOG_ERROR("buf2 is not aligned");
+        return;
+    }
```

Comment:
if(buf2 != NULL)
{
    LV_ASSERT_MSG(buf2 == lv_draw_buf_align(buf2, cf), "buf2 is not aligned");
}

Reply:
You have to add return when failed. Thus no need to use LV_ASSERT_MSG at all.